core/src/main/java/org/keycloak/util/BasicAuthHelper.java

```diff
@@ -18,40 +18,81 @@
 package org.keycloak.util;
 
 import org.keycloak.common.util.Base64;
-import org.keycloak.common.util.Base64Url;
 
+import java.io.IOException;
+import java.io.UnsupportedEncodingException;
+import java.net.URLDecoder;
+import java.net.URLEncoder;
 import java.nio.charset.StandardCharsets;
 
 /**
+ * The default implementation is compliant with <a href="https://datatracker.ietf.org/doc/html/rfc2617">RFC 2617</a>
+ *
  * @author <a href="mailto:bill@burkecentral.com">Bill Burke</a>
  * @version $Revision: 1 $
  */
 public class BasicAuthHelper {
     public static String createHeader(String username, String password) {
-        return "Basic " + Base64.encodeBytes((username + ':' + password).getBytes(StandardCharsets.UTF_8));
+        try {
+            return "Basic " + Base64.encodeBytes((username + ':' + password).getBytes(StandardCharsets.UTF_8), Base64.DO_BREAK_LINES);
```

Comment:
RFC 2617 explicitly says "<base64 encoding of user-pass, except not limited to 76 char/line>".  To me that means that line breaks should not be present.  Ironically that is the way it was before so this change actually appears to have broken RFC 2617 compliance rather than fixed it.

Reply:
Would you mind opening a bug report for this?

Reply:
[Bug report](https://github.com/keycloak/keycloak/issues/24672) created